cmd/dusk/action.go

```diff
@@ -3,35 +3,51 @@ package main
 import (
 	"bytes"
 	"fmt"
-	"math/rand"
-	"os"
-	"os/signal"
-	"time"
-
 	cfg "github.com/dusk-network/dusk-blockchain/pkg/config"
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/message"
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/topics"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/diagnostics"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/logging"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/rpcbus"
-	log "github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"math/rand"
+	"os"
+	"os/signal"
+	"time"
 )
 
-const (
-	// name (wihtout ext) for the config file to look for
-	configFileName = "dusk"
+var (
+	log     *logrus.Entry
+	config  string
+	datadir string
 )
 
-func main() {
+func action(ctx *cli.Context) error {
+
+	// check arguments
+	if arguments := ctx.Args(); len(arguments) > 0 {
+		return fmt.Errorf("failed to read command argument: %q", arguments[0])
+	}
+
+	if datadir = ctx.GlobalString(DataDirFlag.Name); datadir != "" {
```

Comment:
Would you confirm that test harness is compatible with CLI changes?

Reply:
I did not test that, I'll check.

Reply:
it seems to work fine